src/dmd/backend/cod3.d

```diff
@@ -1111,7 +1111,37 @@ static if (NTEXCEPTIONS)
 }
 
         case BCretexp:
-            retregs = regmask(e.Ety, funcsym_p.ty());
+            reg_t reg1, reg2, lreg, mreg;
+            reg1 = reg2 = NOREG;
+            if (config.exe == EX_WIN64) // broken
+                retregs = regmask(e.Ety, funcsym_p.ty());
+            else
+            {
+                retregs = allocretregs(e.Ety, e.ET, funcsym_p.ty(), &reg1, &reg2);
+                assert(reg1 != NOREG || !retregs);
+            }
+
+            lreg = mreg = NOREG;
+            if (reg1 == NOREG)
+            {}
+            else if (tybasic(e.Ety) == TYcfloat)
+                lreg = ST01;
+            else if (mask(reg1) & (mST0 | mST01))
+                lreg = reg1;
+            else if (reg2 == NOREG)
+                lreg = reg1;
+            else if (mask(reg1) & XMMREGS)
+            {
+                lreg = XMM0;
+                mreg = XMM1;
+            }
+            else
+            {
+                lreg = mask(reg1) & mLSW ? reg1 : AX;
+                mreg = mask(reg2) & mMSW ? reg2 : DX;
+            }
+            if (reg1 != NOREG)
+                retregs = (mask(lreg) | mask(mreg)) & ~mask(NOREG);
```

Comment:
What is the purpose of calling allocretregs on line 1120 if it gets redone here?